presto-main/src/main/java/io/prestosql/server/protocol/ExecutingStatementResource.java

```diff
@@ -106,13 +108,15 @@ public ExecutingStatementResource(
             ExchangeClientSupplier exchangeClientSupplier,
             BlockEncodingSerde blockEncodingSerde,
             @ForStatementResource BoundedExecutor responseExecutor,
-            @ForStatementResource ScheduledExecutorService timeoutExecutor)
+            @ForStatementResource ScheduledExecutorService timeoutExecutor,
+            ExecutingStatementResourceConfig executingStatementResourceConfig)
     {
         this.queryManager = requireNonNull(queryManager, "queryManager is null");
         this.exchangeClientSupplier = requireNonNull(exchangeClientSupplier, "exchangeClientSupplier is null");
         this.blockEncodingSerde = requireNonNull(blockEncodingSerde, "blockEncodingSerde is null");
         this.responseExecutor = requireNonNull(responseExecutor, "responseExecutor is null");
         this.timeoutExecutor = requireNonNull(timeoutExecutor, "timeoutExecutor is null");
+        this.defaultTargetResultSize = Optional.of(executingStatementResourceConfig.getServerTargetResultSize());
```

Comment:
Maybe convert `serverTargetResultSize` in `ExecutingStatementResourceConfig.java` to be an `Optional`. That way the code here will be consistent and all future callers will also be able to know the expected behavior regarding nulls.

Reply:
Since the configuration will always produce a value, we don't have to make it optional. I will remove the Optional requirement on defaultTargetResultSize.